### PR TITLE
Extend traces

### PR DESCRIPTION
**Pull request by chriddyp**

### CODE REVIEW

**Comment by chriddyp** [comment]
cc @theengineear @bpostlethwaite

**Comment by chriddyp** [comment]
Usage: http://nbviewer.ipython.org/gist/chriddyp/2fe51027d881076e4f52

File: plotly/widgets/graphWidget.js [hunk 1]
```diff
@@ -14,7 +14,7 @@ require(["widgets/js/widget", "widgets/js/manager"], function (widget, manager)
         // construct the expected IPython3 widget API
         IPYTHON_VERSION = '2';
         manager = {WidgetManager: widget};
-        widget = {DOMWidgetView: IPython.DOMWidgetView}
+        widget = {DOMWidgetView: IPython.DOMWidgetView};
```
Comment (theengineear) [inline plotly/widgets/graphWidget.js]: oops, thx : )

File: plotly/widgets/graph_widget.py [hunk 2]
```diff
@@ -408,7 +408,10 @@ def restyle(self, data, indices=None):
             ```
         """
         # TODO: Add flat traces to graph_objs
-        message = {'task': 'restyle', 'update': data, 'graphId': self._graphId}
+        message = {
+            'task': 'restyle', 'update': update,
+            'graphId': self._graphId
+        }
```
Comment (theengineear) [inline plotly/widgets/graph_widget.py]: This is weird spacing now:

```javascript
message = {
    'task': 'restyle',
    'update': update,
    'graphId': self._graphId
}
```

?

File: plotly/widgets/graph_widget.py [hunk 3]
```diff
@@ -666,3 +669,105 @@ def reorder_traces(self, current_indices, new_indices=None):
...
+                in monitoring applications.
+
+                To set max_points to different values for each trace
+                or attribute, set max_points to a dict mapping keys
+                to max_points values. See the examples below.
+
+            Examples:
+                Initialization - Start each example below with this setup:
+                ```
+                from plotly.widgets import Graph
```
Comment (theengineear) [inline plotly/widgets/graph_widget.py]: `GraphWidget`

File: plotly/widgets/graph_widget.py [hunk 4]
```diff
@@ -666,3 +669,105 @@ def reorder_traces(self, current_indices, new_indices=None):
...
+                        'x': [new_points, new_points],
+                        'y': [new_points, new_points]
+                    },
+                    [0, 1],
+                    max_points={
+                        'x': [25, 50],
+                        'y': [25, 50]
+                    }
+                )
+                ```
```
Comment (theengineear) [inline plotly/widgets/graph_widget.py]: nice examples! in general, i think it'd be helpful to write `indices=[0,1]` where applicable so folks remember what the `[0,1]` is doing.

Also, you should have one example of extending with a '.' attribute, like `marker.size`.

Comment (theengineear) [inline plotly/widgets/graph_widget.py]: along with that, is it worth defaulting indices to be `indices=(0,)` or something?

**Comment by theengineear** [comment]
nothing blocking from my end! just a couple comments. nice!

File: plotly/widgets/graph_widget.py (hunk 4 above)
Comment (chriddyp) [inline plotly/widgets/graph_widget.py]: yeah, that's a good idea.

**Comment by chriddyp** [comment]
cool, just added a couple more examples. all good @theengineear ?

**Comment by theengineear** [comment]
:dancer: Thx!

**Pull request merged**